themes/ddbasic/template.php

```diff
@@ -805,28 +538,22 @@ function ddbasic_remove_default_link_classes($classes) {
 }
 
 /**
- * Allows us to add script plugins to the theme via theme settings.
- *
- * Ex. add a javascript depending on the settings in the theme.
+ * Implements hook_js_alter().
  */
-function ddbasic_load_plugins() {
-  $theme_path = drupal_get_path('theme', 'ddbasic');
-
-  // If sticky menus is enabled in the theme load it.
-  if (theme_get_setting('main_menu_sticky')) {
-
-    // Add variable to js so we can check if it is set.
-    drupal_add_js(array('ddbasic' => array('main_menu_sticky' => theme_get_setting('main_menu_sticky'))), 'setting');
+function ddbasic_js_alter(&$javascript) {
+  // Set the ding_popup.js to the popup-hijack.js instead.
+  $ding_popup = drupal_get_path('module', 'ding_popup') . '/ding_popup.js';
+  if (isset($javascript[$ding_popup])) {
+    $javascript[$ding_popup]['data'] = drupal_get_path('theme', 'ddbasic') . '/scripts/popup-hijack.js';
```

Comment:
`ding_popup` is our own module. Why do we want to replace it in the theme?

If the current implementation provided by ding_popup is no longer needed then it should be replaced. We do not want obsolete code lying around.